tests/testthat/test-routes.R

```diff
@@ -0,0 +1,110 @@
+context("routes")
+
+test_that("URLs are properly redirected with app$redirect", {
+ library(dashHtmlComponents)
+
+ app <- Dash$new()
+
+ app$redirect("/foo", "/")
+ app$redirect("/bar/*", "/")
```

Comment:
Likewise it would be nice if the wildcard was used in the redirect.

Reply:
fixed in 1f5bd7a

Reply:
@alexcjohnson If we're to use parameterized redirects in `app$redirect` as well, I imagine it will require some thought since `keys` exist only within the handler itself. In order for the app developer to just pass in the new path with a key we'd either
- parse the string for `:keys` and loop over them, constructing the new path inside of the handler passed to `app$redirect`
- allow developers to pass in a function (instead of a string), which we would then evaluate to a string in place of `new_path` within `app$redirect`'s internal handler

If it's best to handle this now, just let me know which option makes the most sense. I'd tend to favour the second one if possible, since parsing could get complicated in some cases. Alternatively, I could open an issue to address in a subsequent PR.

Reply:
I like the function option too. I'd do it now, I suspect we're going to need that form sooner rather than later in the docs.

Reply:
@alexcjohnson OK, I've added this additional feature in 4d6c27d. It's somewhat complicated to implement but appears to work as intended. 

I've tried to streamline passing in a function, by ignoring the formals, since the handler *must* have a signature including `(request, response, keys, ...)` anyway, but I'll test with passing in extra arguments as `...` and see if I can get it to work. 

I'm not going to go overboard there, though; if it's not straightforward, passing arbitrary arguments won't be supported in this PR.

I'll also add a unit test for the feature. The current syntax looks like

```
app$redirect("/accounts/:user_id/*", function() paste0("/users/", keys$user_id))
```

Reply:
Documentation for this feature ☝️ added in f6fcbd1.